.eslintrc.cjs

```diff
@@ -0,0 +1,250 @@
+                multiline: true,
+                minProperties: 3,
+            },
+            ExportDeclaration: {                        // ...named exports with at least 3 properties
+                multiline: true,
+                minProperties: 3,
+            },
+        }],
+        'object-curly-spacing': 'error',            // Enforce consistent spacing inside curly brackets
+        '@stylistic/object-property-newline': ['error', {   // Enforce placing object properties on separate lines such that...
+            allowAllPropertiesOnSameLine: true,            // ...properties can be on the same line, with newline-destructuring/newline handling the max number of properties
+        }],
+        'newline-destructuring/newline': 'error',   // Deconstruct object properties on separate lines if objects have more than 3 or more properties
+        'simple-import-sort/imports': 'error',      // Enforce sorting of imports
+        'simple-import-sort/exports': 'error',      // Enforce sorting of exports
+        'import/extensions': ['error', 'ignorePackages', { // Require extensions for imports...
+            ts: 'never',                        // ... except for .ts files since this plugin doesn't allow .js imports in .ts files which is needed for our TS config, but the compiler will catch any import issues anyways
+            js: 'always',                       // ... in .js files for ESM compatibility
+        }],
+
+        // Promises & Async
+        'require-await': 'error',               // Require await in async functions
+
+        // Returns
+        'arrow-body-style': ['error', 'as-needed', {    // Require braces around arrow function bodies...
+            requireReturnForObjectLiteral: true,        // ...especially when returning an object literal
+        }],
+        'no-return-await': 'off',               // Allow return await
+
+        // Miscellaneous
+        'file-progress/activate': 1,            // Show progress in console while linting
+        'no-control-regex': 'off',              // Allow control characters in regular expressions
+    },
+    overrides: [
+        {   // Rules for...
+            // ...TypeScript files only
+            files: [
+                '**/*.ts',
+            ],
+            extends: [
+                'airbnb-typescript/base',
+                'plugin:@stylistic/recommended-extends',
+                'plugin:@stylistic/disable-legacy',
+                'plugin:@typescript-eslint/recommended-requiring-type-checking',
+            ],
+            parser: '@typescript-eslint/parser',
+            parserOptions: {
+                project: [
+                    path.resolve(path.resolve(), 'tsconfig.json'),          // Required for: @typescript-eslint/recommended-requiring-type-checking
+                    path.resolve(path.resolve(), 'tests/tsconfig.json'),    // Required for the below jest overrides to take effect for typescript files
```

Comment:
@beanc16 Does this mean that anyone using this package needs to have their tests in a folder called `tests`? I've seen some projects just call them `test`. If that is the case, we will either need a way to modify/override this easily on the consuming side.

Reply:
I did some thinking on this. This was originally added because it assumes your main `tsconfig.json` doesn't include your test file like so:
```json
{
    ...
    "include": [
        "src",
    ],
    ...
}
```

And then assumes you have a second `tsconfig.json` for your unit tests that's something like this:
```json
{
    ...
    "extends": "../tsconfig.json",
    "include": [
        "**/*.test.ts", // or "**/*.spec.ts", whatever the extension is for their Jest setup
    ],
    ...
}
```

Ultimately, what matters is that your tsconfig includes your test files. With that in mind, I think this would cover the most cases most of the time:
```json
            parserOptions: {
                project: [
                    path.resolve(path.resolve(), 'tsconfig.json'),          // Required for: @typescript-eslint/recommended-requiring-type-checking
                    path.resolve(path.resolve(), 'test/tsconfig.json'),     // Required for the below jest overrides to take effect for typescript files
                    path.resolve(path.resolve(), 'tests/tsconfig.json'),    // Required for the below jest overrides to take effect for typescript files
                ],
            },
```